Added docs for capabilities (#422)
elastic/observability-docs: docs/en/ingest-management/elastic-agent/elastic-agent-capabilities.asciidoc

The above example allows `system/metrics` input and denies every other input in configuration.

With the configuration:

[source,yaml]
----
inputs:
	- type: logfile
	  streams:
	  	- paths: []{a.error,b.access}
	- type: system/logs
	  streams:
	  	- paths: []{a,b}
	- type: system/metrics
	  streams:
	  	- metricset: cpu
	  	- metricset: memory
----

and capabilities as defined above, configuration would result in:

[source,yaml]
----
inputs:
	- type: system/metrics
	  streams:
	  	- metricset: cpu
	  	- metricset: memory
----

In this example, the stream is applied if the host platform is not Windows:

[source,yaml]
----
inputs:
  - type: system/metrics
    streams:
      - metricset: load
        data_stream.dataset: system.cpu
        condition: ${host.platform} != 'windows'
----

In this example, the processor is applied if the host platform is not Windows:

[source,yaml]
----
inputs:
  - type: system/metrics
    streams:
      - metricset: load
        data_stream.dataset: system.cpu
    processors:
      - add_fields:
          fields:
            platform: ${host.platform}
          to: host
        condition: ${host.platform} != 'windows'
----

> **Supported capabilites:**

Reviewer: ```suggestion
[supported-capabilities]
=== Supported capabilities
```

> Capabilities can be used to alter congestion of outputs, inputs or upgrade capability of agent.

Reviewer: ```suggestion
You can use capabilities to modify the congestion of outputs, inputs or upgrade the capability of {agent}.
```

> For inputs we use `input` key as follows containing an expression:

Reviewer: ```suggestion
[capabilities-inputs]
==== Inputs

For inputs, you can the `input` key containing an expression.
```

[source,yaml]
----
version: 0.0.1
capabilities: 
- rule: allow
  input: system/*
----

> For outputs we use `output` key as follows containing an expression:

Reviewer: ```suggestion
[capabilities-outputs]
==== Outputs

For outputs, you can use the `output` key containing an expression.
```

[source,yaml]
----
version: 0.0.1
capabilities: 
- rule: deny
  output: kafka
----

> For upgrade we use `output` key as follows containing an expression allowing upgrade just to version `8.0.0`:

Reviewer: ```suggestion
[capabilities-upgrade]
==== Upgrade

For an upgrade, you can use the `output` key containing an expression that allows a specific upgrade to version `8.0.0`. `${version}` is the target version.
```

[source,yaml]
----
version: 0.0.1
capabilities: 
- rule: allow
  upgrade: "${version} == '8.0.0'"
----

> where `${version}` is a target version.

Reviewer: This can be removed as we addressed in the previous comment.
```suggestion
where `${version}` is a target version.
```

> With version EQL conditions can be used as well as `*` wildcard:

Reviewer: ```suggestion
With a versioned EQL, conditions can be used, as well as a * wildcard. This allows upgrades just for bug fixes.
```

[source,yaml]
----
version: 0.0.1
capabilities: 
- rule: allow
  upgrade: "${version} == '8.0.*'"
----

> allows upgrades just for bug fixes.

Reviewer: This can be removed as we addressed in the previous comment. 
```suggestion
allows upgrades just for bug fixes.
```

> Upgrade capability definition:

Reviewer: ```suggestion
**Upgrade capability definition**

This upgrade definition allows upgrades just for minor version changes.
```

[source,yaml]
----
version: 0.0.1
capabilities: 
- rule: allow
  upgrade: "${version} == '8.*.*'"
----

> Allows upgrades just for minor version change.

Reviewer: This can be removed as we addressed it in the previous comment.

```suggestion
Allows upgrades just for minor version change.
```

> Part of the upgrade action is also a `source_uri` address specifying repository from where artifacts should be
> downloading.

Reviewer: ```suggestion
Part of the upgrade action is also a `source_uri` address that specifies the repository from where artifacts should be
downloaded.
```

> To restrict only to https only we could use:

Reviewer: ```suggestion
To restrict to HTTPS, you can use the following.
```

[source,yaml]
----
version: 0.0.1
capabilities: 
- rule: allow
  upgrade: "startsWith(${source_uri}, 'https')"
----

> Upgrade capability is blocking. This means that in case action does not meet first upgrade definition, we're not proceeding with evaluation.

Reviewer: ```suggestion
In the example above, upgrade capability is blocked. If the action does not meet the first definition, the upgrade is prevented.
```
